### PR TITLE
test(storage): add mrd test for negative offset and 0 length

### PR DESCRIPTION
**Pull request by krishnamd-jkp**

### CODE REVIEW

File: storage/integration_test.go [hunk 1]
```diff
@@ -627,6 +627,88 @@ func TestIntegration_MRDWithReadHandle(t *testing.T) {
 	})
 }
 
+func TestIntegration_MRDWithNegativeOffsetsAndZeroLength(t *testing.T) {
+	multiTransportTest(skipAllButZonal(context.Background(), "Bidi Read API test"), t, func(t *testing.T, ctx context.Context, bucket string, _ string, client *Client) {
+		dataSize := 100
+		// Generate random content for testing.
+		content := make([]byte, dataSize)
+		rand.New(rand.NewSource(0)).Read(content)
+		objName := "MRDWithReadHandle"
```
[the remark this answers is not in the document]
Comment (cpriti-os) [reply storage/integration_test.go:636]: Please do this

File: storage/integration_test.go [hunk 2]
```diff
@@ -627,6 +627,88 @@ func TestIntegration_MRDWithReadHandle(t *testing.T) {
...
+			},
+			{
+				name:   "negative offset positive length",
+				offset: -10,
+				length: 20,
+			},
+			{
+				name:   "negative offset zero length",
+				offset: -10,
+				length: 0,
```
Comment (cpriti-os) [inline storage/integration_test.go:673]: We take care of negative length right?
